tests/integration/db.py

```diff
@@ -70,19 +58,73 @@ class ItemNotFound(Exception):
     """Signals missing DynamoDB Item data"""
 
 
+class DynamoDBResource(threading.local):
+    def __init__(self, **kwargs):
+        conf = kwargs
+        if not conf.get("endpoint_url"):
+            if os.getenv("AWS_LOCAL_DYNAMODB"):
+                conf.update(
+                    dict(
+                        endpoint_url=os.getenv("AWS_LOCAL_DYNAMODB"),
+                        aws_access_key_id="Bogus",
+                        aws_secret_access_key="Bogus",
+                    )
+                )
+        # If there is no endpoint URL, we must delete the entry
+        if "endpoint_url" in conf and not conf["endpoint_url"]:
+            del conf["endpoint_url"]
+        if "region_name" in conf:
+            del conf["region_name"]
+        region = conf.get("region_name", os.getenv("AWS_DEFAULT_REGION", "us-east-1"))
+        self.conf = conf
+        self._resource = boto3.resource(
+            "dynamodb", config=botocore.config.Config(region_name=region), **self.conf
+        )
+
+    def __getattr__(self, name):
+        return getattr(self._resource, name)
+
+    def get_latest_message_tablenames(
+        self, prefix: str = "message", previous: int = 1
+    ) -> list[str]:
+        """Fetches the name of the last message table"""
+        client = self._resource.meta.client
+        paginator = client.get_paginator("list_tables")
+        tables = []
+        for table in paginator.paginate().search(
+            "TableNames[?contains(@,'{}')==`true`]|sort(@)[-1]".format(prefix)
+        ):
+            if table and table.encode().startswith(prefix):
+                tables.append(table)
+        if not len(tables) or tables[0] is None:
+            return [prefix]
+        tables.sort()
+        return tables[0 - previous :]
+
+    def get_latest_message_tablename(self, prefix: str = "message") -> str:
+        """Fetches the name of the last message table"""
+        return self.get_latest_message_tablenames(prefix=prefix, previous=1)[0]
+
+
+class DynamoDBTable(threading.local):
+    def __init__(self, ddb_resource: DynamoDBResource, *args, **kwargs) -> None:
+        self._table = ddb_resource.Table(*args, **kwargs)
+
+    def __getattr__(self, name):
+        return getattr(self._table, name)
+
+
 # Autopush utils:
 def generate_hash(key: bytes, payload: bytes) -> str:
     """Generate a HMAC for the uaid using the secret
 
     :returns: HMAC hash and the nonce used as a tuple (nonce, hash).
```

Comment:
Heh, I thought that function docstrings end with a blank line, but I see neither [PEP8](https://peps.python.org/pep-0008/#documentation-strings) and [PEP257](https://peps.python.org/pep-0257/#multi-line-docstrings) do that. 

TIL.

Reply:
I admit that some linting magic must have removed those lines.